src/java.base/share/classes/java/math/BigInteger.java

```diff
@@ -347,16 +347,17 @@ public class BigInteger extends Number implements Comparable<BigInteger> {
      * @since 9
      */
     public BigInteger(byte[] val, int off, int len) {
-        if (val.length == 0) {
+        Objects.checkFromIndexSize(off, len, val.length);
```

Comment:
Does this change which exception get thrown if multiple pre-conditions are violated? That is okay, but if that is the case, the issue should get a quick CSR for the behavioral change.